package/crds/ssoadmin.aws.upbound.io_permissionsets.yaml

```diff
@@ -95,6 +95,7 @@ spec:
                     description: Key-value map of resource tags.
                     type: object
                 required:
+                - instanceArn
```

Comment:
This is a breaking change. It's a consequence of normalizing the external name for this resource.

From a practical point of view, this will only affect Observe only resources, because this parameter was always required by terraform, and therefore included in the kubernetes validation. If I make it no longer required, if this is missing, and the external name annotation is set to the permission set id (the normalized external name I chose), crossplane will be unable to construct the terraform id, which contains the instance arn, so it won't be able to import the existing resource.

I could remove it from the `IdentifierFields` in the external name configuration, which would avoid the breaking schema change, but it feels like it really _should_ be required, especially since omitting it will cause imports to fail. 

Another option would be to change the external name to include the instance arn (perhaps by changing it from the short id of the permission set to the full ARN of the permission set, which embeds the id of the instance, from which the instance arn can be derived). But that seems like sacrificing the long-term user experience in order to avoid a minor breaking api schema change.

Reply:
With this removed from `IdentifierFields` in the external name config, the following examples still work:
* A resource that specifies all the fields in `spec.forProvider` with default management policies
* A resource that doesn't include this field in `spec.forProvider`, but does include it in `spec.initProvider`, with default management policies (this field is immutable, so there's really no reason someone would *want* to put it in `spec.initProvider`)
* A resource with only the `Observe` management policy, that specifies the external name annotation and populates this field to the correct value in `spec.forProvider`

This doesn't work:
* A resource with only the `Observe` management policy, that specifies the external name annotation, but doesn't populate this field. This passes the kubernetes validator, but fails at runtime because it can't construct the terraform id.

Reply:
I can think of a handful of ways forward here:

* Normalize the external name to just the provider-generated part of the ARN, and leave `instanceArn` as an identifier field. If this was a new resource, as opposed to an update to an existing one, this seems like it would be the clear winner. 
* Normalize the external name, but don't specify `instanceArn` as an identifier field. This will cause an error for Observe resources as mentioned above.
* Use the permission set ARN as the external name (as opposed to the current external name which is `${permission_set_arn},${instance_arn}`. This could be done without specifying `instanceArn` as an identifier field because the instance arn can be derived from the permission set arn. 
* Leave the current external name which matches the terraform id. This would certainly be the smallest change, but it's also a fairly cumbersome external name to have to use. 

One factor that seems relevant to deciding is what happens to existing managed resources during a provider upgrade, when the external name config changes. I'm hoping that as long as the resources were ready, crossplane would simply reconstruct the external name from the terraform id, using the new logic, which would provide the correct result, but I'm not sure quite how best to test this. Would it be sufficient to take a MR that has become ready and simply edit its external-name annotation to the old format? Would there be other edits I should do at the same time? Or is it necessary to do a "full" test, where I actually do a provider upgrade on my test cluster?